start.sh

```diff
@@ -20,12 +10,30 @@ dpkg-reconfigure xserver-xorg-legacy
 if [[ -z ${FLAGS+x} ]]
   then
     echo "Using default chromium flags"
-    export FLAGS=" $KIOSK --disable-dev-shm-usage --ignore-gpu-blacklist --enable-gpu-rasterization --force-gpu-rasterization --autoplay-policy=no-user-gesture-required --user-data-dir=/usr/src/app/settings --enable-features=WebRTC-H264WithOpenH264FFmpeg"
+    export FLAGS="--disable-dev-shm-usage --autoplay-policy=no-user-gesture-required --noerrdialogs"
+
+    # if DISABLE_GPU is NOT set, add the GPU flags
+    if [[ ! -z ${DISABLE_GPU+x} ]] && [[ "$DISABLE_GPU" -eq "1" ]]
+      then
+        # don't add the GPU flags
+        echo "Disabling GPU acceleration"
+      else
+        echo "Enabling GPU acceleration"
+        FLAGS="$FLAGS --enable-features=WebRTC-H264WithOpenH264FFmpeg --ignore-gpu-blacklist --enable-gpu-rasterization --force-gpu-rasterization --gpu-sandbox-failures-fatal=no --enable-native-gpu-memory-buffers"
+    fi
+fi
+
+# if the PERSISTENT enVar is set, add the appropriate flag
+if [[ ! -z $PERSISTENT ]] && [[ "$PERSISTENT" -eq "1" ]]
+  then
+    echo "Adding user settings directory"
+    FLAGS="$FLAGS --user-data-dir=/data"
 fi
```

Comment:
just confirming if the primitive is used in a multicontainer application, the user will have to create a named volume and bind mount to `/data` for that service?

Reply:
Yes, that's the plan. Once I have this built and pushed to Docker, I'll branch balenaDash, use the primitive and check that this works OK.